### PR TITLE
actions for large numbers of incidents

### LINKED ISSUE #156: Batch large numbers of action invocations by API limits

### PR DESCRIPTION
**Pull request by martindstone**
to fix #156

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,3 @@
 # MUST match .node-version
-nodejs 18.16.1
+nodejs 20.4.0
```
Comment (gsreynolds) [inline .tool-versions:2]:
```suggestion
nodejs 18.16.1
```

Won't yet be able to move to Node 20 (will tackle via #145)

File: src/util/incidents.js [hunk 2]
```diff
@@ -58,9 +58,11 @@ export const generateIncidentActionModal = (incidents, action) => {
...
   }
-  const primaryMessage = `${i18next.t('Incidents')}(s) ${unresolvedIncidents
-    .map((i) => i.incident_number)
-    .join(', ')} ${i18next.t('have been')} ${i18nAction}.`;
+  let primaryIncidentList = unresolvedIncidents.slice(0, 100).map((i) => i.incident_number).join(', ');
+  if (unresolvedIncidents.length > 100) {
+    primaryIncidentList += ` ${i18next.t('and')} ${unresolvedIncidents.length - 100} ${i18next.t('more')}`;
+  }
+  const primaryMessage = `${i18next.t('Incidents')} ${primaryIncidentList} ${i18next.t('have been')} ${i18nAction}.`;
   const secondaryMessage = `(${resolvedIncidents.length} ${i18next.t('Incidents')} ${i18next.t(
```
Comment (gsreynolds) [inline src/util/incidents.js:66]: Will need to adjust this, as we need to translate the entire phrase and not individual words.

Comment (martindstone) [reply src/util/incidents.js:66]: Do you have any pointers on how to do that? I still haven't learned enough about how we do translations

Comment (gsreynolds) [reply src/util/incidents.js]: Just pushed a commit with that. It's using https://www.i18next.com/translation-function/interpolation. There are a few examples in the src/locales folders. 

Essentially:

translation key of 
```
`${i18next.t('Incidents X have been ACTIONED', { primaryIncidentList, i18nAction })}.`
```
(I've just been using capitalised placeholders as a convention)

and then in the locale files, translate with the interpolation:
```
  "Incidents X have been ACTIONED": "Incidents {{primaryIncidentList}} ont été {{i18nAction}}",
```

Often the word order can be different, which is why it needs to be the entire phrase.

Comment (martindstone) [reply src/util/incidents.js]: Ohhhh ok that seems pretty easy... I will try it

**Review by gsreynolds — APPROVED** [review]

**Pull request merged**